client/src/pages/challenges.js

```diff
@@ -30,95 +33,129 @@ export default withStyles({
                 <label class='form-ext-label' for='check1'>Show Solved</label>
               </div>
             </div>
           </div>
         </div>
         <div class='col-6'>
           {
-            problems.map(problem => {
-              const hasDownloads = problem.files.length !== 0
-
-              const error = errors[problem.id]
-              const hasError = error !== undefined
-
-              return (
-                <div class={`frame ${classes.frame}`} key={problem.id}>
-                  <div class='frame__body'>
-                    <div class='row u-no-padding'>
-                      <div class='col-6 u-no-padding'>
-                        <div class='frame__title title'>{problem.category}/{problem.name}</div>
-                        <div class='frame__subtitle u-no-margin'>{problem.author}</div>
-                      </div>
-                      <div class='col-6 u-no-padding u-text-right'>
-                        <div class='frame__subtitle faded' style='margin-top: .75rem; margin-bottom: 0'>{problem.points.max} pts</div>
-                      </div>
-                    </div>
-
-                    <div class='content-no-padding u-center'><div class={`divider ${classes.divider}`} /></div>
-                    <div class='frame__subtitle'>{problem.description}</div>
-                    <form class='form-section' onSubmit={this.submitFlag(problem.id)}>
-
-                      {
-                        hasError &&
-                          <label class='text-danger info font-light'>{error}</label>
-                      }
-                      <div class='form-group'>
-                        <input class={`form-group-input input-small ${hasError ? 'input-error' : ''}`} placeholder='Flag' value={values[problem.id]} onChange={this.linkState('values.' + problem.id)} />
-                        <button class='form-group-btn btn-small'>Submit</button>
-                      </div>
-                    </form>
-
-                    {
-                      hasDownloads &&
-                        <div>
-                          <p class='faded frame__subtitle u-no-margin'>Downloads</p>
-                          <div class='tag-container'>
-                            {
-                              problem.files.map(file => {
-                                return (
-                                  <div class='tag' key={file.path}>
-                                    <a href={config.staticEndpoint + '/' + file.path}>
-                                      {file.name}
-                                    </a>
-                                  </div>
-                                )
-                              })
-                            }
-
-                          </div>
-                        </div>
-                    }
-                  </div>
-                </div>
-              )
-            })
+            showSolved
```

Comment:
This should probably not be inline, it's a bit hard to read. Maybe move it above the return statement?

Reply:
With the current method of rendering each problem, making this its own function is a bit messy. Feel free to change if you think its better.

Reply:
Pushed a fix moving it above the return statement